src/App.tsx

```diff
@@ -14,11 +16,82 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  showList: boolean;
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    showList: false,
+  };
+
+  reverseList = () => {
+    this.setState(state => ({ goods: [...state.goods].reverse() }));
```

Comment:
```suggestion
    this.setState({ goods } => { goods: [...goods].reverse() });
```
Здесь и в дальнейшем можно сразу делать деструктуризацию и принимать только нужные свойства, но тут как кому легче в понимании. Если так понятнее, то лучше не трогать)